target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-eap620hd-v2.dts

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/dts-v1/;
+
+#include "ipq6018.dtsi"
+#include "ipq6018-cp-cpu.dtsi"
+#include "ipq6018-ess.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "TP-Link EAP620HD V2";
+	compatible = "tplink,eap620hd-v2", "qcom,ipq6018";
+
+	aliases {
+		serial0 = &blsp1_uart3;
+		led-boot = &led_status_blue;
+                led-failsafe = &led_status_blue;
+                led-running = &led_status_blue;
+                led-upgrade = &led_status_blue;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs-append = " ubi.block=0,rootfs root=/dev/ubiblock0_1";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 9 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+                led_status_blue: status-blue {
+                        function = LED_FUNCTION_STATUS;
+                        gpios = <&tlmm 35 GPIO_ACTIVE_HIGH>;
+                        color = <LED_COLOR_ID_BLUE>;
+                };
+
+
+	};
+
+	gpio-restart {
```

Comment:
Is this used anywhere? Maybe just remove this node, not quite useful. You can always use PSCI to reset the system.

Reply:
To the extent that a GPIO is connected, we'd want that represented in the device tree. How we use it, or whether we use it at all does not affect the requirement to describe it. I suggest keeping the node.

In this specific case, binging the GPIO prevents one from accidentally rebooting the system from userspace by inadvertently setting this pin.

Reply:
If this actually resets the board then it should be modeled like it is